src/rgw/rgw_d3n_datacache.cc

```diff
@@ -55,14 +53,19 @@ int D3nCacheAioWriteRequest::create_io(bufferlist& bl, unsigned int len, string
 D3nDataCache::D3nDataCache()
   : index(0), cct(NULL), io_type(ASYNC_IO), free_data_cache_size(0), outstanding_write_size(0)
 {
-  tp = new D3nL2CacheThreadPool(32);
+  lsubdout(g_ceph_context, rgw_datacache, 5) << "D3nDataCache: " << __func__ << "()" << dendl;
+  if( g_conf()->rgw_d3n_l2_distributed_datacache_enabled ) {
+    lsubdout(g_ceph_context, rgw_datacache, 5) << "D3nDataCache: " << __func__ << "(): rgw_d3n_l2_distributed_datacache_enabled" << dendl;
+    tp = new D3nL2CacheThreadPool(32);
+  }
 }
 
 int D3nDataCache::io_write(bufferlist& bl, unsigned int len, std::string oid)
 {
   D3nChunkDataInfo* chunk_info = new D3nChunkDataInfo;
+  std::string location = cache_location + oid;
 
-  std::string location = cache_location + oid; /* replace tmp with the correct path from config file*/
+  lsubdout(g_ceph_context, rgw_datacache, 20) << "D3nDataCache: " << __func__ << "(): location=" << location << dendl;
   FILE *cache_file = 0;
```

Comment:
Used to be iso c++, but c++11 should assign nullptr, right?  Don't care about "File *cache_file" as this is basically C...

Reply:
ack